### ISSUE TITLE
Form validation errors are not read by some screen readers

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3562]
## Describe the bug

Error messages are not announced by some screen reader/browser combinations. This is because the current implementation does not dynamically write content to an existing live region but instead adds the live region and the message it contains to the DOM all at once. 

## How to reproduce

https://clarity.design/documentation/stepper

- Using Firefox/NVDA or IE11/NVDA or EDGE/JAWS, activate the 'Stepper Demos' link under the 'Stepper' heading
- Trigger an error message by using the Tab key to focus the 'First Name' field and then press the Tab key again to move focus to the next control in the Tab order.
- Observe that an error message appears below the 'First Name' field and after it does, the screen reader does not announce it.

## Expected behavior

The error should be announced.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

The aria-live property indicates that an element will be updated. So it should be in the DOM when the page loads and subsequently updated.

When the form is visible on the page, the live region should be on the page as well and empty as seen below.

<clr-control-error aria-live="polite">
</clr-control-error>

Then, as errors occur, the live region should be populated as seen below.

<clr-control-error aria-live="polite">
Error: The first name field does not have a valid entry
</clr-control-error>

See https://www.w3.org/TR/wai-aria-1.1/#aria-live

**Comment by gnomeontherun** [comment]
@coryrylan Can we close this as done then?

**Comment by coryrylan** [comment]
I don't think so, I think this might be still an issue and needs to be verified.

**Comment by clane** [comment]
Updated recommendation - use an ARIA Live Region to inform the user the application is in an error state and use aria-describedby to programmatically associate the inline error messages with their respective fields. Than focus should be set to to the first field in error.  See the example below.

```
<div aria-live="polite">There are errors ...</div>

<label for="nameField">name</name>
<input id="name" aria-describedby="inlineError1"/>
<div id="inlineError1">The name field should not contain integers</div>
```

To recap, the live region is used to perform a status update that tells the user the form is in an error state. Then focus is set to the first field in error. Since aria-describedby is used to associate the errors with their respective input fields, screen readers should announce the errors when the fields gain focus.

**Comment by bdryanovski** [comment]
I think that this is solved now with the new AriaLiveService. At least with the version that supports `clr-control-error` component. 

There is no need to change the focus to another element - we could use & we are using the `AriaLiveService.announce` to broadcast these types of messages.

```ts 
// src/clr-angular/forms/common/error.ts

 ngAfterViewInit() {
    this.ariaLiveService.announce(this.el.nativeElement);
  }
```

so in a way when the error is first to render on the screen it will be broadcasted.

But it's left to the developer to decide what types of errors are shown and how they are grouped. For example, `clr-controll-error` could be used for general error for the form or single error by fields.

**Comment by bdryanovski** [comment]
When I tested the Stepper demo when I get an error the SR is announcing that there is an error but also found this two cases and i'm not sure are they OK.

* when both fields have error (first name & last name) the SR read one of the errors twice. (not sure why)I think that the form-error component was created and destroyed for some reason, this could be just bad implemented demo.

* when focusing on the input the SR did not announce that there is an error right now!

**Comment by gnomeontherun** [comment]
Ok let's not focus on edge cases first. Did we confirm on our supported platforms that it does read out the error state the first time? If there are more complex requirements, it becomes an application issue to resolve due to our lack of control over the entire form.

If you focus on a field with an error, aria-describedby should read out the message if it is present. There might not be a message, so that would be a demo problem.

**Comment by bdryanovski** [comment]
I have to make some additional tests and will come back with some demo. What I'm thinking that we only have to add `describedby` but have to test it. Fields and error blocks are separate components if I remember correctly.

**Comment by bdryanovski** [comment]
I'm closing this we are deprecating our internal use of ClrAriaLiveService so the code that was previously buggy and trying to read and announce the form input error messages is no longer there. 

To handle this the user/developer will have to use the ClrAriaLiveService in his wrapper component and decide what and when will be the error message that will be announced to the viewer by calling the `ClrAriaLiveSerivce.announce('message')`.
